Jenkinsfile

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env groovy
+
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+
+node('java') {
+    timestamps {
+        stage('Checkout') {
+            checkout scm
+        }
+
+        stage('Build') {
+            withEnv([
+                "JAVA_HOME=${tool 'jdk8'}",
+                "PATH+MVN=${tool 'mvn'}/bin",
+                'PATH+JDK=$JAVA_HOME/bin',
+            ]) {
+                timeout(30) {
```

Comment:
Is that not the correct behavior? Taking more than 30 minutes sounds like a problem with tests taking a long time. If this is going to spin for 60 minutes and still fail then that's costing us double the amount of compute $$$ :frowning: